src/api/RequestApi.ts

```diff
@@ -84,24 +99,113 @@ const getMyRequests = async () => {
-      return Promise.reject([] as IInRequest[]);
+      return Promise.reject([]);
     } else {
-      const response: IResponseItem[] = await spWebContext.web.lists
+      try {
+        return spWebContext.web.lists
+          .getByTitle("Items")
+          .items.filter(
+            `supGovLead/Id eq '${userId}' or employee/Id eq '${userId}'`
+          )
+          .select(requestedFields)
+          .expand(expandedFields)();
+      } catch (e) {
+        console.error(
+          `Error occurred while trying to fetch requests for user ${userId}}`
+        );
+        console.error(e);
+        if (e instanceof Error) {
+          throw new ApiError(
+            e,
+            `Error occurred while trying to fetch requests for user with ID ${userId}: ${e.message}`
+          );
+        } else if (typeof e === "string") {
+          throw new ApiError(
+            new Error(
+              `Error occurred while trying to fetch requests for user with ID ${userId}: ${e}`
+            )
+          );
+        } else {
+          throw new ApiError(
+            undefined,
+            `Unknown error occurred while trying to fetch requests for user with ID ${userId}`
+          );
+        }
+      }
+    }
+  }
+};
+
+const getRequest = async (Id: number) => {
+  if (process.env.NODE_ENV === "development") {
+    let response = testItems[Id - 1];
+    return Promise.resolve(response);
+  } else {
+    try {
+      return spWebContext.web.lists
         .getByTitle("Items")
-        .items.filter(
-          `supGovLead/Id eq '${userId}' or employee/Id eq '${userId}'`
-        )
+        .items.getById(Id)
         .select(requestedFields)
         .expand(expandedFields)();
-      return response.map((request) => transformInRequestFromSP(request));
+    } catch (e) {
+      console.error(
+        `Error occurred while trying to fetch request with ID ${Id}}`
+      );
+      console.error(e);
+      if (e instanceof Error) {
+        throw new ApiError(
+          e,
+          `Error occurred while trying to fetch request with ID ${Id}: ${e.message}`
+        );
+      } else if (typeof e === "string") {
+        throw new ApiError(
+          new Error(
+            `Error occurred while trying to fetch request with ID ${Id}: ${e}`
+          )
+        );
+      } else {
+        throw new ApiError(
+          undefined,
+          `Unknown error occurred while trying to fetch request with ID ${Id}`
+        );
+      }
     }
   }
 };
 
+const getRequests = async () => {
+  if (process.env.NODE_ENV === "development") {
+    let response = testItems;
+    return Promise.resolve(response);
+  } else {
+    return spWebContext.web.lists
+      .getByTitle("Items")
+      .items.select(requestedFields)
+      .expand(expandedFields)();
+  }
+};
+
+// Exported hooks for working with requests
+
 export const useMyRequests = () => {
+  return useQuery(["requests", "currentUser"], getMyRequests, {
+    select: transformInRequestsFromSP,
```

Comment:
Ok, I see what you are doing here.  This is what you were thinking we should do on the RolesApi, where we transform on the select after the useQuery has cached, rather than caching the transformed data?